integration/test_magic_folder.py

```diff
@@ -156,31 +220,57 @@ def test_bob_creates_alice_deletes_alice_restores(magic_folder):
     # both alice and bob make a file at "the same time".
     alice_dir, bob_dir = magic_folder
 
-    # really, we fudge this a little: in reality, either alice or bob
-    # "wins" by uploading to the DMD first. So we make sure bob wins
-    # this one by giving him a massive head start
+    # either alice or bob will "win" by uploading to the DMD first.
     with open(join(bob_dir, 'alpha'), 'w') as f:
         f.write("this is bob's alpha\n")
-    time.sleep(1.0)
     with open(join(alice_dir, 'alpha'), 'w') as f:
         f.write("this is alice's alpha\n")
 
-    # since bob uploaded first, alice should see a backup
-    util.await_file_contents(join(alice_dir, 'alpha'), "this is bob's alpha\n")
-    util.await_file_contents(join(alice_dir, 'alpha.backup'), "this is alice's alpha\n")
+    # there should be a conflict, but we don't know who will win the
+    # race
+    acceptable = [
+        join(bob_dir, 'alpha.conflict'),
+        join(alice_dir, 'alpha.conflict'),
+    ]
+    found = None
+    start_time = time.time()
+    while not found and time.time() - start_time < 15.0:
+        print("  waiting for conflict")
+        print("bob")
+        print(listdir(bob_dir))
+        print("alice")
+        print(listdir(alice_dir))
+        for path in acceptable:
+            if exists(path):
+                print("FOUND {}".format(path))
+                found = path
+                break
+        time.sleep(1)
 
-    util.await_file_contents(join(bob_dir, 'alpha'), "this is alice's alpha\n")
-    util.await_file_contents(join(bob_dir, 'alpha.backup'), "this is bob's alpha\n")
+    assert found, "should have gotten a conflict"
+    if exists(acceptable[0]):  # bob got the conflict, so alice won the race
+        assert open(join(bob_dir, 'alpha'), 'r').read() == "this is bob's alpha\n"
+        assert open(join(bob_dir, 'alpha.conflict'), 'r').read() == "this is alice's alpha\n"
+        assert open(join(alice_dir, 'alpha'), 'r').read() == "this is alice's alpha\n"
+        if exists(join(alice_dir, 'alpha.conflict')):
+            assert open(join(alice_dir, 'alpha.conflict'), 'r').read() == "this is bob's alpha\n"
+        # hmm, so Alice is getting a conflict *too*? Is that okay?
+        # assert not exists(join(alice_dir, 'alpha.conflict'))
+    else:                      # alice got the conflict, so bob won the race
+        assert exists(acceptable[1])
+        assert open(join(alice_dir, 'alpha'), 'r').read() == "this is alice's alpha\n"
+        assert open(join(alice_dir, 'alpha.conflict'), 'r').read() == "this is bob's alpha\n"
+        assert open(join(bob_dir, 'alpha'), 'r').read() == "this is bob's alpha\n"
+        if exists(join(bob_dir, 'alpha.conflict')):
+            assert open(join(bob_dir, 'alpha.conflict'), 'r').read() == "this is alice's alpha\n"
+        # hmm, so Bob is getting a conflict *too*? Is that okay?
+        # assert not exists(join(bob_dir, 'alpha.conflict'))
```

Comment:
Maybe these two branches would be better off setting up some data and then allowing all of the assertions to follow in one unified codepath.

Also.  It does *not* seem okay that both sides get a conflict.

Reply:
Maybe my "one side 'wins'" logic is incorrect, and it's really "both sides lose" because they each see an update they're not "expecting"..?